src/scirpy/tl/_mutational_load.py

```diff
@@ -0,0 +1,278 @@
+from collections.abc import Sequence
+from typing import Literal, Union
+
+import pandas as pd
+
+from scirpy.get import airr as get_airr
+from scirpy.util import DataHandler
+
+
+def simple_hamming_distance(sequence: str, germline: str, frequency=False):
+    if (sequence is None) or (germline is None):
+        return None
+    if len(sequence) != len(germline):
+        raise Exception("Sequences might not be IMGT aligned")
+
+    distance = 0
+    if frequency:
+        frac_counter = len(germline)
+        for i, c in enumerate(germline):
+            if (c == ".") or (c == "N"):
+                frac_counter -= 1
+                continue
+            elif (sequence[i] == ".") or (sequence[i] == "N"):
+                frac_counter -= 1
+                continue
+            elif c != sequence[i]:
+                distance += 1
+        if frac_counter == 0:
+            return frac_counter
+        else:
+            return distance / frac_counter
+
+    else:
+        for i, c in enumerate(germline):
+            if (c == ".") or (c == "N"):
+                continue
+            if (sequence[i] == ".") or (sequence[i] == "N"):
+                continue
+            elif c != sequence[i]:
+                distance += 1
+
+        return distance
```

Comment:
I think my implementation for the hamming distance might not be ideal in this case, since it's meant to create a matrix to compare all sequences to each other. The simple_hamming_distance however is mainly used to compare (a list of) pairs and also needs to ignore certain characters. This use case scales linearly with the input size, so maybe the performance is not a problem anyway.

Reply:
ok, then let's not optimize that for now. If it becomes a bottleneck we can think of applying some numba to the function or optimize it in any other way.